### ISSUE TITLE
TX interpolation >= 16x problem

### ISSUE THREAD

**Issue by srcejon** [f4exb/sdrangel#1895]
As was discussed on #1892, the M17 modulator did not appear to work with low baseband sample rate (<100k).

I suspect the problem is actually with interpolation when set to 16x or greater.

Here's a picture of TX and RX spectrum of the packet mod, with Int = 8 - which looks OK.

![image](https://github.com/f4exb/sdrangel/assets/57259258/363f467a-68d3-4c3d-a6b9-df7dd5e9562a)

And here it is with Int = 16

![image](https://github.com/f4exb/sdrangel/assets/57259258/bdf36278-a6a5-467b-9c97-a9bd21d3c050)

Which looks like it has a problem.

**Comment by f4exb** [comment]
"MyB210" suggests you are using a USRP here. I have tried with a HackRF Tx + RTL-SDR Rx combo and I do not notice the issue:

https://github.com/f4exb/sdrangel/assets/6192319/6a074c5d-d7ab-48f6-bb5c-7510129091fe

Also what modulator do you use? I have used the packet modulator in a loop transmission to obtain similar pulses.

**Comment by srcejon** [comment]
Yep, a USRP with with Packet Mod.

In usrpoutputthread.h

       Interpolators<qint16, SDR_TX_SAMP_SZ, 12> m_interpolators;

In hackrfoutputthread:

      Interpolators<qint8, SDR_TX_SAMP_SZ, 8> m_interpolators;

So different data-types / scaling may explain it.

**Comment by srcejon** [comment]
BladeRF uses the same interpolators, but I don't have one.

Also, might need something with higher res ADC than RTL SDR to see the problem.

**Comment by f4exb** [comment]
Yes I just tried with the BladeRF 2 at an interpolation of 16 and using the Aaronia RTSA as the receiver which cannot suspected of bad dynamic range. Even with a dynamic of 60dB on the spectrum no arifacts are visible.

![image](https://github.com/f4exb/sdrangel/assets/6192319/039fc3e8-4930-4cb3-81f4-1ae469bb2179)

**Comment by srcejon** [comment]
Ok, thanks. Will try to narrow it down some more here.

**Comment by sp5wwp** [comment]
Are there any updates to this?

**Comment by srcejon** [comment]
Working on something else at the moment, I'm afraid.

**Comment by sp5wwp** [comment]
Is this bug still present?

**Comment by srcejon** [comment]
I suspect so, unfortunately.
